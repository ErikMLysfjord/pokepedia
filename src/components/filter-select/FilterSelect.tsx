import { ChangeEvent } from "react";
import "./filter-select.css";

const Capitalize = (str: string) => {
  return str.charAt(0).toUpperCase() + str.slice(1);
};

const FilterSelect = ({
  options,
  selected,
  handleChange,
  label,
}: {
  options: string[];
  selected: string;
  handleChange: (e: ChangeEvent<HTMLSelectElement>) => void;
  label: string;
}) => {
  return (
    <div className="app__filtering-container">
      <span className="app__filtering-text">{label}</span>
      <select
        value={selected}
        onChange={handleChange}
        className="filter-select"
        placeholder="Select a color"
      >
        {options.map((option, index) => {
          if (option === "none") {
            return (
              <option key={`${option}-${index}`} value={option}>
                Select a color
              </option>
            );
          }
          return (
            <option key={`${option}-${index}`} value={option}>
              {Capitalize(option)}
            </option>
          );
        })}
      </select>
    </div>
  );
};

export default FilterSelect;
