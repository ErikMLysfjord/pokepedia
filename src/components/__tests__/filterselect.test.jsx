import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import FilterSelect from "../filter-select/FilterSelect";
import { fireEvent } from "@testing-library/dom";

describe("FilterSelect", () => {
  test("should match snapshot", () => {
    const { container } = render(
      <FilterSelect
        options={["test", "test2"]}
        handleChange={(e) => {
          console.log(e.target.value);
        }}
        selected="test"
      />
    );

    expect(container).toMatchSnapshot();
  });

  test("should render correctly", () => {
    render(
      <FilterSelect
        options={["none", "test", "test2"]}
        handleChange={(e) => {
          console.log(e.target.value);
        }}
        selected="test"
      />
    );

    expect(screen.getByText("Test")).toBeInTheDocument();
    expect(screen.getByText("Select a color")).toBeInTheDocument();
    expect(screen.getByRole("combobox")).toBeInTheDocument();
    expect(screen.getByRole("combobox")).toHaveValue("test");
    expect(screen.getAllByRole("option")).toHaveLength(3);
  });

  test("should change option", () => {
    render(
      <FilterSelect
        options={["test", "test2"]}
        handleChange={(e) => e.target.value}
        selected="test"
      />
    );

    const select = screen.getByRole("combobox");

    fireEvent.select(select, { target: { value: "test2" } });
    expect(screen.getByRole("combobox")).toHaveValue("test2");
  });
});
