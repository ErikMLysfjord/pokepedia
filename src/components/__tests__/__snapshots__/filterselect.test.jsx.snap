// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterSelect > should match snapshot 1`] = `
<div>
  <select
    class="filter-select"
    placeholder="Select a color"
  >
    <option
      value="test"
    >
      Test
    </option>
    <option
      value="test2"
    >
      Test2
    </option>
  </select>
</div>
`;

exports[`FilterSelect > should render 1`] = `
<div>
  <select
    class="filter-select"
    placeholder="Select a color"
  >
    <option
      value="test"
    >
      Test
    </option>
    <option
      value="test2"
    >
      Test2
    </option>
  </select>
</div>
`;
